tox.c

```diff
@@ -324,33 +324,37 @@ static _Bool init_avatar(AVATAR *avatar, const char_t *id, uint8_t *png_data_out
     }
     return 0;
 }
-
 static size_t load_save(uint8_t **out_data){
     uint8_t path[UTOX_FILE_NAME_LENGTH], *p, *data;
-    uint32_t size;
+    char savename[UTOX_FILE_NAME_LENGTH];
 
+    uint32_t size;
+    const unsigned int tox_savename_len = strlen(tox_savename);
     do{ /* Try the STS compliant save location */
         p = path + datapath(path);
-        strcpy((char*)p, "tox_save.tox");
+        memcpy(savename, tox_savename, tox_savename_len);
+        memcpy(savename+tox_savename_len, ".tox", sizeof(".tox")); /* Add tox extension */
+        memcpy((char*)p, savename, tox_savename_len+sizeof(".tox"));
         data = file_raw((char*)path, &size);
         if(data) break; /* We have data, were done here! */
         /* Try filename missing the .tox extension */
         p = path + datapath(path);
-        strcpy((char*)p, "tox_save");
```

Comment:
@SmoothDude on line 339, do you want to support profiles without the .tox extension? how are you accounting for if the user uses `--profile someuser.tox` instead of `--profile someuser`